convert/convert.go

```diff
@@ -61,6 +75,52 @@ func Convert(inputFilename, outputFilename string, from, to Format) error {
 	return nil
 }
 
+func convertKongGateway2xTo3x(input *file.Content, filename string) (*file.Content, error) {
+	if input == nil {
+		return nil, fmt.Errorf("input content is nil")
+	}
+	outputContent := input.DeepCopy()
+
+	convertedRoutes := []*file.FRoute{}
+	for _, service := range outputContent.Services {
+		for _, route := range service.Routes {
+			convertedRoutes = append(convertedRoutes, migrateRoutesPathFieldPre300(route, filename))
+		}
+		service.Routes = convertedRoutes
+	}
+
+	if outputContent.FormatVersion != "3.0" {
+		cprint.UpdatePrintf(
+			"From the '%s' config file, the _format_version field has been migrated from '%s' to '%s'.\n"+
+				"These automatic changes may not be correct or exhaustive enough, please\n"+
+				"perform a manual audit of the config file.\n\n"+
+				"For related information, please visit: https://docs.konghq.com/deck/latest/3.0-upgrade\n",
+			filename, outputContent.FormatVersion, "3.0")
```

Comment:
I'll recommend throwing the warning unconditionally.

Reply:
https://github.com/Kong/deck/pull/753/commits/6347de40814d87c4ad9d9365b01202b315c640c4